leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/PresenceStatus.java

```diff
@@ -28,7 +30,9 @@
     private Presence state;
 
     /* Elements to handle the time the client waits before going to sleep */
-    private Timer clientAwakeTimer;
+    ScheduledExecutorService clientAwakeExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Create 1 scheduler by device sounds so good.
This is more about creating 1 scheduler for all tasks in PresenceService.

Reply:
But then you need to specify the number of tasks when creating the sheduler, and we don't now how many clients will be registered at a time... that's why I took this approach.

Reply:
This is not the number of tasks, this is more the number of workers.
A scheduler have 1 or severals workers/thread.
You can schedule more tasks than workers available.
The only risk is that task could be delay a bit.

Reply:
Ok, I've misunderstood it. So then with just one single thread executor we can handle all the timer tasks of the registered clients, right?

Reply:
Yep